crates/node-builder/src/components/traits.rs

```diff
@@ -142,3 +155,37 @@ where
         self(ctx)
     }
 }
+
+/// Trait for launching the node.
+pub trait LaunchNode<DB, Types, Components>
```

Comment:
can we add a blanket impl for Fn that returns a future, similar to:

https://github.com/paradigmxyz/reth/blob/d6dd1c2dabbc1801c1b68da357bf7fcaff4a47a0/crates/node-builder/src/components/payload.rs#L20-L26

Reply:
great idea!